codebuild/bin/s2n_open_fds_test.py

```diff
@@ -0,0 +1,81 @@
+import os
+import re
+import sys
+
+# Allow stdin, stdout, stderr, and the file descriptor for dynamic analysis log to remain open at exit
+ACCEPTABLE_OPEN_FDS = 4
+ERROR_EXIT_CODE = 123
+
+analysis_file_location = "../../build/Testing/Temporary"
+analysis_file_pattern = re.compile(r"^LastDynamicAnalysis.*")
+# This regular expression captures valgrind 3.13 and valgrind 3.18+ log
+fd_pattern = re.compile(r"FILE DESCRIPTORS: \d+ open(?: \(\d+ std\))? at exit.$")
+error_message_start_pattern = re.compile(r"^Running .*/s2n_.*_test.c.*")
+error_message_end_pattern = re.compile(r"^<end of output>$")
+
+
+def open_analysis_file(path):
+    for file_name in os.listdir(path):
+        if analysis_file_pattern.match(file_name):
+            file = open(os.path.join(path, file_name), 'r')
+    return file
+
+
+def print_error_message(file_name, error_message):
+    # Take the path to the failing test file
+    print(f"{file_name.split(' ')[1]} leaks file descriptor(s):\n")
+    print(''.join(error_message))
+    return 0
+
+
+def read_analysis_file(file):
+    exit_code = 0
```

Comment:
Is there any particular reason for declaring all of these variables at the start of the python function? It looks like very old school C to me, haha. I think it's more idiomatic to declare variables closer to their point of use?

Reply:
> Is there any particular reason for declaring all of these variables at the start of the python function?

I want to use them to store information about parsing. I have changed the logic of this test, so there are only two variables that I need.